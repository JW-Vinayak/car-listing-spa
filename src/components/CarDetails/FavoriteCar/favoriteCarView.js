import React, { useState, Fragment } from "react";
import * as localStorage from "../../../utils/localStorage";
import PropTypes from "prop-types";

const FavoriteCarSetting = props => {
  let { stockNumber } = props;
  let key = stockNumber.toString();
  let [isStored, setIsStored] = useState(localStorage.hasCarAsFavorite(key));
  let content;

  let removeItem = () => {
    localStorage.removeCarAsFavorite(key);
    setIsStored(false);
  };

  let setItem = () => {
    localStorage.setCarAsFavorite(key);
    setIsStored(true);
  };
  if (!isStored) {
    content = (
      <Fragment>
        <p>
          if you like this car, click the button and save it in your collection
          of favourite items.
        </p>
        <button className="button favorite" onClick={() => setItem()}>
          Save
        </button>
      </Fragment>
    );
  } else {
    content = (
      <Fragment>
        <p>
          if you wish, you can click the button and remove this car from your
          collection of favourite items.
        </p>
        <button className="button favorite" onClick={() => removeItem()}>
          Remove
        </button>
      </Fragment>
    );
  }

  return content;
};

FavoriteCarSetting.propTypes = {
  stockNumber: PropTypes.number.isRequired
};

export default FavoriteCarSetting;
