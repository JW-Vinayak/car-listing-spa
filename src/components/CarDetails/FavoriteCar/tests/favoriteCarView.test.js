import React from "react";
import FavoriteCar from "../favoriteCarView";
import renderer from "react-test-renderer";

describe("Favorite car component", () => {
  it("shallow renders without crashing", () => {
    const tree = renderer.create(<FavoriteCar stockNumber={1234} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
