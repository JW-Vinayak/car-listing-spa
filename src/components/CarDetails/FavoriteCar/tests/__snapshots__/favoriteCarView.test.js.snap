// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Favorite car component shallow renders without crashing 1`] = `
Array [
  <p>
    if you like this car, click the button and save it in your collection of favourite items.
  </p>,
  <button
    className="button favorite"
    onClick={[Function]}
  >
    Save
  </button>,
]
`;
